src/realm/object-store/schema.cpp

```diff
@@ -250,6 +250,11 @@ static void compare(ObjectSchema const& existing_schema, ObjectSchema const& tar
             changes.emplace_back(schema_change::RemoveProperty{&existing_schema, &current_prop});
             continue;
         }
+        if (current_prop.type_is_nested() != target_prop->type_is_nested() ||
```

Comment:
I think you should compare nesting levels and not just if the type is nested.

Reply:
Isn't the following check already cattering for this? `current_prop.nested_types != target_prop->nested_types`